blog/macos-rig/index.qmd

```diff
@@ -0,0 +1,309 @@
+
+For your rig configuration, you would create a similar script, edit the `.Rprofile`, and restart the session.
+
+11. Configure [RStudio](https://github.com/gvelasq/dotfiles/tree/main/rstudio).
+
+[Snippets](https://support.posit.co/hc/en-us/articles/204463668-Code-Snippets-in-the-RStudio-IDE) help quickly insert common short pieces of code. Open RStudio’s snippet config for R scripts:
+
+```r
+usethis::edit_rstudio_snippets("r")
+```
+
+Append [this script](https://github.com/gvelasq/dotfiles/blob/main/rstudio/r.snippets) to the bottom of the file for some useful examples.
+
+:::{.callout-note}
+Change the prefix of your snippets to your initials. The default in these examples is `gv`.
+:::
+
+Try a snippet out! After appending the script to your snippet file, type `dml` in the Console and press tab:
+
+![](gif1.gif){fig-align="center"}
+
+For whenever you want to remember De Morgan's laws 😉
+
+Now, open RStudio's preference file by running:
+
+```r
+usethis::edit_rstudio_prefs()
+```
+
+Add the content of [this file](https://github.com/gvelasq/dotfiles/blob/main/rstudio/rstudio-prefs.json) to the `rstudio-prefs.json` file. This file configures RStudio to look and act the way that you’d like (rainbow parentheses, anybody? 🌈).
+
+:::{.callout-note}
+* Change `default_project_location` to your preferred directory. The default in this example is `~/code`.
+* Change `git_exe_path` to the location of your Git executable, which can be found by typing `which git` in a terminal. The default in this example is `/usr/local/bin/git`.
+* Change document_author to your name. The default in this example is `gvelasq`.
+:::
+
+Finally, add the contents of [`rstudio_bindings.json`](https://github.com/gvelasq/dotfiles/blob/main/rstudio/rstudio_bindings.json) to the local version found in this hidden folder (I don’t think there’s a {usethis} command for this, unfortunately!):
+
+```bash
+~/.config/rstudio/keybindings/rstudio_bindings.json
+```
+
+Now, anytime you want to make sure you're using the most current version of RStudio, you just have to run <kbd>Shift</kbd>+<kbd>Cmd</kbd>+<kbd>R</kbd>!
+
+12. Set Git credentials in R.
+
+The {[gitcreds](https://gitcreds.r-lib.org/)} package allows you to set your GitHub token once and use it everywhere (Git, R, and the RStudio IDE). Use it to set up RStudio with a [Personal Access Token](https://happygitwithr.com/https-pat.html#store-pat) (PAT) with GitHub and have it work every time. 
+
+Install the {gitcreds} package:
+
+```r
+install.packages("gitcreds")
+```
+ Manage your GitHub tokens or passwords:
```

Comment:
This is actually a two-step procedure so this section needs to be reworded. First, `usethis::create_github_token()` will opens a browser window to the GitHub form to generate a PAT, with suggested scopes pre-selected' -- the first part is key because it's helpful, the last part is key because it helps you make the right choice of scopes every time. Then, the command will suggest to use `gitcreds::gitcreds_set()` to save the PAT locally using the macOS Git credential manager (after copying it from GitHub).